CryptoPkg/Library/OpensslLib/OpensslLib.inf

```diff
@@ -666,9 +666,6 @@
   DebugLib
   RngLib
 
-[LibraryClasses.ARM]

```

Comment:
nit: Should we also remove ARM from the list of supported archs for each of these infs? Or leave it since parts of OpenSSL can be used on ARM?

Reply:
Yeah, good question. The problem is that OpensslLib is a big pile of related but distinct functionality, and normally (on Linux), it is distributed as separate libcrypto.so and libssl.so shared libraries.

So for ARM, this means that OpensslLib might still be used, as long as the user does not pull in functionality that requires floating point support. Given that MbedTlsLib depends on OpensslLib too (!) -for SM3 support-, we actually still have to provide an OpensslLib resolution on 32-bit ARM too - it will just never reference the objects that require floating point support.

I would prefer for MbedTlsLib not to depend on OpensslLlib at all - the SM3 support case may be important, and I can see how hacking up support this way is acceptable at the product level, I strongly feel that the upstream EDK2 should be held to higher standards in this regard. If MbedTls does not support SM3, and SM3 is a requirement, just use OpensslLib and be done with it.

So TL;DR, I think we should remove this once we drop the OpensslLib dependence from MbedTlsLib.